js/state/syncUtil.js

```diff
@@ -485,20 +571,22 @@ module.exports.createSiteData = (site, appState) => {
   let value
   if (module.exports.isSyncable('bookmark', immutableSite)) {
     name = 'bookmark'
+    const isFolder = bookmarkFoldersUtil.isFolder(immutableSite)
+    const sitesCollection = isFolder ? STATE_SITES.BOOKMARK_FOLDERS : STATE_SITES.BOOKMARKS
     objectId = site.objectId ||
       folderToObjectMap[site.folderId] ||
-      module.exports.newObjectId(['sites', siteKey])
+      module.exports.newObjectId([sitesCollection, siteKey])
     parentFolderObjectId = site.parentFolderObjectId ||
       folderToObjectMap[site.parentFolderId] ||
       findOrCreateFolderObjectId(site.parentFolderId, appState)
     value = {
       site: siteData,
-      isFolder: siteUtil.isFolder(immutableSite),
+      isFolder,
       hideInToolbar: site.parentFolderId === -1,
       parentFolderObjectId
     }
   } else if (siteUtil.isHistoryEntry(immutableSite)) {
```

Comment:
@ayumi can we remove this function? This function is not relevant anymore. So if we needed it we need to write a new one in historyUtil file.